### PR TITLE
Fix background color and text color for a badge label in nav bar

### PR DESCRIPTION
**Pull request by vickprat**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes
How the badge is going to look in different scenarios on icons present in the navigation bar?
**Navigation Bar with Primary Style**

Light Mode                                                
Background Color in badge: White                                               
Text Color in badge: App theme color

Dark Mode
Background Color in badge: App theme color
Text color in badge: White

**Navigation Bar with System style**

Both Light and Dark Mode
Background color in badge: Red                 
Text color in badge: White                                      

### Verification (Dark and Light Modes)
**Screenshots where the navigation bar is of primary style**
|Screenshot 2021-11-02 at 5 38 36 PM|Screenshot 2021-11-02 at 5 43 11 PM|

**Screenshots where the navigation bar is of system style**
|Screenshot 2021-11-09 at 6 25 13 PM|Screenshot 2021-11-09 at 6 24 48 PM|

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [x] VoiceOver and Keyboard Accessibility
- [x] Internationalization and Right to Left layouts
- [x] Different resolutions (1x, 2x, 3x)
- [x] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)

### CODE REVIEW

File: ios/FluentUI/Label/BadgeLabel.swift
Comment (rdeassis) [inline ios/FluentUI/Label/BadgeLabel.swift:40]: Did the design team approve this specific variation?
Seems a bit odd that both the background and the badge are white.

Screen Shot 2021-11-08 at 11 50 07 AM

File: ios/FluentUI/Label/BadgeLabel.swift
Comment (rdeassis) [inline ios/FluentUI/Label/BadgeLabel.swift:37]: This could be a guard let to early return in case the property is false.

Comment (vickprat) [reply ios/FluentUI/Label/BadgeLabel.swift]: Done!!

File: ios/FluentUI/Label/BadgeLabel.swift
Comment (vickprat) [reply ios/FluentUI/Label/BadgeLabel.swift:39]: Fixed!! It was an issue in my code itself and so there was no need of designers discussion.

**Review by rdeassis — COMMENTED** [review]
As we spoke offline, there's an issue in the regular title case that the toggle switch does not do anything regarding the badge on the buttons. Can you make sure this is fixed in a separate PR? (hiding or disabling the toggle):

MicrosoftTeams-image

File: ios/FluentUI/Label/BadgeLabel.swift
Comment (rdeassis) [inline ios/FluentUI/Label/BadgeLabel.swift:36]: nit: can you put the shouldUseWindowColor first? then it doesn't need to access the window property if it's not needed.

**Comment by harrieshin** [comment]
hey, I think the red badge should be the default, and then we should just have a way for client team to change the background color of the badge whatever they want? similar to how badge in tabbar could work.

File: ios/FluentUI/Label/BadgeLabel.swift
Comment (vickprat) [reply ios/FluentUI/Label/BadgeLabel.swift]: Done!!

File: ios/FluentUI/Navigation/BadgeLabelButton.swift
Comment (vickprat) [reply ios/FluentUI/Navigation/BadgeLabelButton.swift]: Now, I made it non-optional but just wanted to know that what is the problem to make this var optional?

If the intention is to make the caller forcibly set this value then should I go ahead and change the initializer itself? But again do we really need to do that?

**Comment by vickprat** [comment]
> hey, I think the red badge should be the default, and then we should just have a way for client team to change the background color of the badge whatever they want? similar to how badge in tabbar could work.

Hi Harrie!! We generallly follow the window color thing as we do in case of tooltip also because there is background color both in light and dark mode and also the text color in both light and dark mode. There is an email thread started by me and suggested by Rafael on this and designers expect some other design changes as well. Let me forward that email thread to you.

Also, because this PR is a blocker for shipping the notification center feature and this week is the last week because next week will be code freeze so I am planning to improve the design as suggested in that email thread as a separate CR on FluentUI.

**Comment by vickprat** [comment]
> As we spoke offline, there's an issue in the regular title case that the toggle switch does not do anything regarding the badge on the buttons. Can you make sure this is fixed in a separate PR? (hiding or disabling the toggle):
> 
> MicrosoftTeams-image

I have fixed the bug and this was happening with the rainbow on avatar thing too. This is the PR: https://github.com/microsoft/fluentui-apple/pull/793

File: ios/FluentUI/Navigation/BadgeLabelButton.swift
Comment (rdeassis) [reply ios/FluentUI/Navigation/BadgeLabelButton.swift]: The rationale behind that is that we're already providing a default boolean value for that, so if the calling code is changing the value, it needs to be a valid boolean (instead of a possible nil value).

**Review by rdeassis — APPROVED** [review]
Signing off provided that @harrieshin is okay with not going with the custom color suggested for now.
@vickprat can you just confirm before we go ahead and merge this?

**Comment by rdeassis** [comment]
@vickprat I'm merging this PR to unblock your work. We're expecting the follow up changes to make this conform to the new design guidelines as well as addressing @harrieshin 's [comment](https://github.com/microsoft/fluentui-apple/pull/785#issuecomment-964453453) about the custom colors (below):

`hey, I think the red badge should be the default, and then we should just have a way for client team to change the background color of the badge whatever they want? similar to how badge in tabbar could work.`

**Pull request merged**
